diff --git a/Initialize-Script.ps1 b/Initialize-Script.ps1
deleted file mode 100644
index 27271bf..0000000
--- a/Initialize-Script.ps1
+++ /dev/null
@@ -1,11 +0,0 @@
-param (
-    [String]$inputPath = "$env:homeshare\VDI-UserData\Download\generic\inputs\",
-    [String]$fileName = "mailbox_list.csv",
-    [String]$user = $null
-)
-
-if ($user) {
-    $allMailboxes = @($user)
-} else {
-    $allMailboxes = Get-Content "$inputPath\$fileName"
-}
\ No newline at end of file
diff --git a/Initializers.ps1 b/Initializers.ps1
new file mode 100644
index 0000000..38740fa
--- /dev/null
+++ b/Initializers.ps1
@@ -0,0 +1,34 @@
+function Initialize-DefaultParams {
+    param (
+        [Parameter(Mandatory = $true)]
+        [ValidateNotNullOrEmpty()]
+        [String]$scriptName,
+
+        [String]$inputPath = "$env:homeshare\VDI-UserData\Download\generic\inputs\",
+        [String]$inputFilename = "input.csv",
+
+        [String]$outputPath = "$env:homeshare\VDI-UserData\Download\generic\outputs\",
+        [String]$outputDir = $null,
+        [String]$outputFilename = "output.csv"
+
+        # [ValidateNotNullOrEmpty()]
+        # [String[]]$mailboxes = (Get-Content "$inputPath\$inputFilename")
+    )
+
+    # TODO: Find a way to make this work
+    $cStack = @(Get-PSCallStack)
+    $script = $cStack[$cStack.Length-1].InvocationInfo.MyCommand
+
+    Write-Host "scriptName: $($script)"
+
+    return @{
+        inputPath = $inputPath;
+        inputFilename = $inputFilename;
+
+        outputPath = $outputPath;
+        outputDir = $outputDir;
+        outputFilename = $outputFilename;
+
+        mailboxes = $mailboxes;
+    }
+}
diff --git a/Resolve-PendingMigrations.ps1 b/Resolve-PendingMigrations.ps1
index ee68998..79e7745 100644
--- a/Resolve-PendingMigrations.ps1
+++ b/Resolve-PendingMigrations.ps1
@@ -19,7 +19,7 @@ $routingAddress = (
 
 foreach ($mailbox in $allMailboxes) {
     $itemCount = Get-MailboxStatistics $mailbox | Select-Object -ExpandProperty ItemCount
-    if ($itemCount -le "300") {
+    if ($itemCount -le "400") {
         $mailboxInfo = Get-Mailbox -Identity $mailbox
         $mailboxInfo.EmailAddresses > $outputPath\$mailbox.txt
         $hasArchive = ($mailboxInfo.archiveGuid -ne "00000000-0000-0000-0000-000000000000") -and $mailboxInfo.archiveDatabase
diff --git a/Set-MailboxQuota.ps1 b/Set-MailboxQuota.ps1
index aed594f..dd5d885 100644
--- a/Set-MailboxQuota.ps1
+++ b/Set-MailboxQuota.ps1
@@ -1,19 +1,14 @@
-# param (
-#     [String]$inputPath = "$env:homeshare\VDI-UserData\Download\generic\inputs\",
-#     [String]$fileName = "mailbox_list.csv",
-#     [String]$user = $null
-# )
-
-# if ($user) {
-#     $allMailboxes = @($user)
-# } else {
-#     $allMailboxes = Get-Content "$inputPath\$fileName"
-# }
-
-# Write-Host "Script:" $PSCommandPath
-# Write-Host "Path:" $PSScriptRoot
-
-. "$PSScriptRoot\Initialize-Script.ps1" -user $user
+param (
+    [String]$inputPath = "$env:homeshare\VDI-UserData\Download\generic\inputs\",
+    [String]$fileName = "mailbox_list.csv",
+    [String]$user = $null
+)
+
+if ($user) {
+    $allMailboxes = @($user)
+} else {
+    $allMailboxes = Get-Content "$inputPath\$fileName"
+}
 
 $sizeFieldName = "TotalItemSizeInGB"
 
diff --git a/Test.ps1 b/Test.ps1
new file mode 100644
index 0000000..01d78b4
--- /dev/null
+++ b/Test.ps1
@@ -0,0 +1,6 @@
+. "$PSScriptRoot\Initializers.ps1"
+$params = Invoke-Expression "Initialize-DefaultParams -scriptName Test $args"
+
+Write-Output "name: $($MyInvocation.MyCommand.Name)"
+Write-Output "name2: $($MyInvocation.ScriptName)"
+Write-Output "inputPath: $($params.inputPath)"
